ccp/src/cu/proving_thread/async_/thread.rs

```diff
@@ -127,34 +122,32 @@ impl ProvingThreadFacade for ProvingThreadAsync {
     ) -> Result<(), Self::Error> {
         let message = NewCCJob::new(dataset, flags, epoch, cu_id);
         let message = AsyncToSyncMessage::NewCCJob(message);
-        self.inlet.send(message).await.map_err(Into::into)
+        self.to_sync.send(message).await.map_err(Into::into)
     }
 
     async fn pin(&self, core_id: LogicalCoreId) -> Result<(), Self::Error> {
         let message = AsyncToSyncMessage::PinThread(PinThread { core_id });
-        self.inlet.send(message).await.map_err(Into::into)
+        self.to_sync.send(message).await.map_err(Into::into)
     }
 
     async fn pause(&mut self) -> Result<(), Self::Error> {
         let message = AsyncToSyncMessage::Pause;
-        self.inlet.send(message).await?;
+        self.to_sync.send(message).await?;
 
-        match self.outlet.recv().await {
+        match self.from_sync.recv().await {
             Some(SyncToAsyncMessage::Paused) => Ok(()),
-            Some(message) => Err(ProvingThreadError::channel_error(format!(
+            Some(message) => Err(AsyncThreadError::channel_error(format!(
                 "expected the Paused event, but {message:?} received"
             ))),
-            None => Err(ProvingThreadError::channel_error(
+            None => Err(AsyncThreadError::channel_error(
                 "sync to async channel is closed unexpectedly".to_string(),
             )),
         }
     }
 
     async fn stop(self) -> Result<(), Self::Error> {
         let message = AsyncToSyncMessage::Stop;
-        self.inlet.send(message).await?;
-        self.sync_thread.join()?;
-
-        Ok(())
+        self.to_sync.send(message).await?;
+        self.sync_thread.join().map_err(Into::into)
```

Comment:
I think `?` syntax should be used here.

This would work too:
```
Ok(self.sync_thread.join()?)
```

but explicit `Ok(())` is even more idiomatic.

Reply:
I think it is abuse only because `()` is returned.  When some sensible value is returned, map_err is more or less Ok, but `Ok(...?)` is preferred.

Reply:
Yep, for short one line expression it's a better option. Let's stick with it.